tests/ui/filter_methods.rs

```diff
@@ -1,24 +1,48 @@
 #![warn(clippy::all, clippy::pedantic)]
 #![allow(clippy::missing_docs_in_private_items)]
+#![allow(clippy::redundant_closure)]
+#![allow(clippy::unnecessary_filter_map)]
 
 fn main() {
-    let _: Vec<_> = vec![5; 6].into_iter().filter(|&x| x == 0).map(|x| x * 2).collect();
+    let options: Vec<Option<i8>> = vec![Some(5), None, Some(6)];
+    let results = ["1", "lol", "3", "NaN", "5"];
 
-    let _: Vec<_> = vec![5_i8; 6]
-        .into_iter()
-        .filter(|&x| x == 0)
-        .flat_map(|x| x.checked_mul(2))
+    // validate iterator of values triggers filter_map + map lint
+    let _: Vec<_> = vec![5_i8; 6].into_iter()
+        .filter_map(|x| x.checked_mul(2))
+        .map(|x| x.checked_mul(2))
         .collect();
 
-    let _: Vec<_> = vec![5_i8; 6]
-        .into_iter()
-        .filter_map(|x| x.checked_mul(2))
-        .flat_map(|x| x.checked_mul(2))
```

Comment:
Can you leave these tests in and add a comment, that this won't get linted anymore?

Reply:
Only one of the removed tests isn't covered by one of the replacement tests as far as I can tell and I added it back in per your request.